src/wh_server_counter.c

```diff
@@ -0,0 +1,104 @@
+/*
+ * Copyright (C) 2024 wolfSSL Inc.
+ *
+ * This file is part of wolfHSM.
+ *
+ * wolfHSM is free software; you can redistribute it and/or modify
+ * it under the terms of the GNU General Public License as published by
+ * the Free Software Foundation; either version 3 of the License, or
+ * (at your option) any later version.
+ *
+ * wolfHSM is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License
+ * along with wolfHSM.  If not, see <http://www.gnu.org/licenses/>.
+ */
+/*
+ * src/wh_server_counter.c
+ */
+#include <string.h>
+#include "wolfhsm/wh_server.h"
+#include "wolfhsm/wh_server_counter.h"
+#include "wolfhsm/wh_message.h"
+#include "wolfhsm/wh_packet.h"
+#include "wolfhsm/wh_error.h"
+
+int wh_Server_HandleCounter(whServerContext* server, uint16_t action,
+    uint8_t* data, uint16_t* size)
+{
+    int ret;
+    uint32_t counter;
+    whPacket* packet = (whPacket*)data;
+    whNvmMetadata meta[1] = {0};
+
+    if (server == NULL || server->nvm == NULL || data == NULL || size == NULL)
+        return WH_ERROR_BADARGS;
+
+    switch (action)
+    {
+    case WH_COUNTER_RESET:
+        /* write 0 to nvm with the supplied id and user_id */
+        counter = 0;
+        memset((void*)meta, 0, sizeof(meta));
+        meta->id = MAKE_WOLFHSM_KEYID(WOLFHSM_KEYTYPE_COUNTER,
+            server->comm->client_id, packet->counterResetReq.keyId);
+        meta->len = sizeof(counter);
+        ret = wh_Nvm_AddObject(server->nvm, meta, meta->len,
```

Comment:
I thought we were going to store the counter value in the label so we didn't need any data segment?